src/components/wifi_scan.rs

```diff
@@ -171,56 +172,44 @@ impl WifiScan {
             let networks = tokio_wifiscanner::scan().await;
             match networks {
                 Ok(nets) => {
-                    let mut wifi_nets: Vec<WifiInfo> = Vec::new();
                     let now = Local::now();
-                    for w in nets {
-                        if let Some(n) = wifi_nets.iter_mut().find(|item| item.ssid == w.ssid) {
-                            let signal: f32 = w.signal_level.parse().unwrap_or(-100.00);
-                            if n.signal < signal {
-                                n.signal = signal;
-                                n.mac = w.mac;
-                                let channel = w.channel.parse::<u8>().unwrap_or(0);
-                                n.channel = channel;
-                            }
-                        } else {
-                            wifi_nets.push(WifiInfo {
-                                time: now,
-                                ssid: w.ssid.clone(),
-                                channel: w.channel.parse::<u8>().unwrap_or(0),
-                                signal: w.signal_level.parse::<f32>().unwrap_or(-100.00),
-                                mac: w.mac.clone(),
-                                color: COLORS_NAMES[wifi_nets.len() % COLORS_NAMES.len()],
-                            });
+                    let wifi_nets: Vec<WifiInfo> = nets.iter().map(|w| {
+                        WifiInfo {
+                            time: now,
+                            ssid: w.ssid.to_string(),
+                            channel: w.channel.parse::<u8>().unwrap_or(0),
+                            signal: w.signal_level.parse::<f32>().unwrap_or(-100.00),
+                            mac: w.mac.to_string(),
+                            color: COLORS_NAMES[nets.len() % COLORS_NAMES.len()],
```

Comment:
This if nice BUT you removed whole logic behind detecting duplicated reports for same WiFi network.